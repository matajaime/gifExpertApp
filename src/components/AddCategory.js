import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({addCategory}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        //console.log(event);
        if ( inputValue != '' ) {
            addCategory(inputValue);
            setInputValue('');    
        }
    };

    const cleanInput = (event) => {
        event.target.value = '';
    }

    return (
        <form onSubmit={handleSubmit}>
        <h4>Add Category</h4>
        <input type="text" value={inputValue} placeholder="Ingresa una categoria" 
        onChange={handleInputChange} onFocus={cleanInput}/>
        </form>
    );
};

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired
}
