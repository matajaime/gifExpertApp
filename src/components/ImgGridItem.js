import React from "react";
import PropTypes from 'prop-types';

export const ImgGridItem = ({img}) => {
    //console.log(img);
    return (
        <div className="card animate__animated animate__fadeInDownBig">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    );
};

ImgGridItem.propTypes = {
    img : PropTypes.object.isRequired
}

