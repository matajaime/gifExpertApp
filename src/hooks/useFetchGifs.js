import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });
    useEffect(()=>{
        getGifs(category).then((imgs)=>{
            setState({
                data: imgs,
                loading: false
            });
        });
        //console.log('Me llamaron');
    },[category]);
    return state;
};

