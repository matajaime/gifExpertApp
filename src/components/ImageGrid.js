import React, {useEffect, useState} from "react";
import { ImgGridItem } from "./ImgGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const ImageGrid = ({category}) => {
    
    //const[gifs, setGifs] = useState([]);
    //useEffect( () => {
    //    getGifs(category).then( setGifs );
    //}, [ category ]);

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
        <h3>{category}</h3>
        { loading ? <p>Cargando...</p> : 
        <div className="card-grid">
            { images.map((gif)=>{
                return <ImgGridItem key={gif.id} img={gif} />;
            })}
        </div>}
        </>
    )
};

ImageGrid.propTypes = {
    category: PropTypes.string.isRequired
}
