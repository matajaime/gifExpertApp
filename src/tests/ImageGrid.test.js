import React from 'react';
import { shallow } from 'enzyme';
import { ImageGrid } from '../components/ImageGrid';
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock("../hooks/useFetchGifs");

describe('ImgGrid tests', ()=>{
    const category = 'Mario';
    test('SnapShot of init state of Grid',() => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true    
        });
        const wrapper = shallow(<ImageGrid key={category} category={category} />);
        expect(wrapper).toMatchSnapshot();
        const pLoading = wrapper.find('p');
        expect( pLoading.exists() ).toBe( true );
    });
    test('snapshot of grid populated', ()=>{
        const imgs = [{
            id: 1,
            url: 'https://test.domain.com/test/path/img.jpg',
            title: 'test title'
        },{
            id: 2,
            url: 'https://test.domain.com/test/path/img2.jpg',
            title: 'test title 2'
        }];
        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false    
        });
        const wrapper = shallow(<ImageGrid key={category} category={category} />);
        expect(wrapper).toMatchSnapshot();
        const pLoading = wrapper.find('p');
        expect( pLoading.exists() ).toBe( false );
        const gridItems = wrapper.find('ImgGridItem');
        expect( gridItems.length ).toBe( imgs.length );
    });
});