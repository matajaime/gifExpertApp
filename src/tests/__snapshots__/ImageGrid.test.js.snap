// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgGrid tests SnapShot of init state of Grid 1`] = `
<Fragment>
  <h3>
    Mario
  </h3>
  <p>
    Cargando...
  </p>
</Fragment>
`;

exports[`ImgGrid tests snapshot of grid populated 1`] = `
<Fragment>
  <h3>
    Mario
  </h3>
  <div
    className="card-grid"
  >
    <ImgGridItem
      img={
        Object {
          "id": 1,
          "title": "test title",
          "url": "https://test.domain.com/test/path/img.jpg",
        }
      }
      key="1"
    />
    <ImgGridItem
      img={
        Object {
          "id": 2,
          "title": "test title 2",
          "url": "https://test.domain.com/test/path/img2.jpg",
        }
      }
      key="2"
    />
  </div>
</Fragment>
`;
